.github/actions/set_branch_name/action.yml

```diff
@@ -0,0 +1,16 @@
+name: Set BRANCH_NAME environment variable
+description: Sets the BRANCH_NAME env variable based on the Github Event
+permissions: {}
+
+runs:
+    using: composite
+    steps:
+        - name: Get branch name (merge)
+          if: github.event_name != 'pull_request'
+          shell: bash
+          run: echo "BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/} | tr / -)" >> $GITHUB_ENV
```

Comment:
In the non-branch case, `GITHUB_REF_NAME` should work, right?

Reply:
ah yes true. yes that would simplify this.